plugins/vendor/services/ZoomVendorService.php

```diff
@@ -0,0 +1,431 @@
+	}
+
+
+	/**
+	 * @action submitRegistration
+	 * @param string $defaultUserId
+	 * @param string $zoomCategory
+	 * @param string $accountId
+	 * @return string
+	 * @throws PropelException
+	 */
+	public function submitRegistrationAction($defaultUserId, $zoomCategory, $accountId)
+	{
+		$partnerId = kCurrentContext::getCurrentPartnerId();
+		$zoomIntegration = VendorIntegrationPeer::retrieveSingleVendorPerPartner($accountId,
+			VendorTypeEnum::ZOOM_ACCOUNT, $partnerId);
+		if(is_null($zoomIntegration))
+		{
+			$zoomIntegration = new VendorIntegration();
+			$zoomIntegration->setAccountId($accountId);
+			$zoomIntegration->setVendorType(VendorTypeEnum::ZOOM_ACCOUNT);
+			$zoomIntegration->setPartnerId($partnerId);
+		}
+		$zoomIntegration->setStatus(VendorStatus::ACTIVE);
+		$zoomIntegration->setDefaultUserEMail($defaultUserId);
+		$zoomIntegration->setZoomCategory($zoomCategory);
+		$zoomIntegration->save();
+		return true;
+	}
+
+	/**
+	 * @action recordingComplete
+	 * @throws Exception
+	 */
+	public function recordingCompleteAction()
+	{
+		KalturaLog::info('Zoom - upload entry to Kaltura starter');
+		$this->verifyHeaderToken();
+		myPartnerUtils::resetAllFilters();
+		$request_body = file_get_contents(self::PHP_INPUT);
+		$data = json_decode($request_body, true);
+		list($accountId, $downloadToken, $hostEmail, $downloadURL, $meetingId) = $this->extractDataFromRecordingCompletePayload($data);
+		$zoomIntegration = VendorIntegrationPeer::retrieveSingleVendorPerPartner($accountId, VendorTypeEnum::ZOOM_ACCOUNT);
+		$emails = $this->extractCoHosts($meetingId, $zoomIntegration, $accountId);
+		// user logged in - need to re-init kPermissionManager in order to determine current user's permissions
+		$ks = null;
+		$dbUser = kuserPeer::getKuserByPartnerAndUid($zoomIntegration->getPartnerId(), $hostEmail, true);
+		if (!$dbUser) //if not go to default user
+		{
+			$emails[] = $hostEmail;
+			$dbUser = kuserPeer::getKuserByPartnerAndUid($zoomIntegration->getPartnerId(), $zoomIntegration->getDefaultUserEMail(), true);
+		}
+		kSessionUtils::createKSessionNoValidations($dbUser->getPartnerId() , $dbUser->getPuserId() , $ks, 86400 , false , "" , '*' );
+		kCurrentContext::initKsPartnerUser($ks);
+		kPermissionManager::init();
+		$url = $this->parseDownloadUrl($downloadURL, $downloadToken);
+		$entryId = $this->createEntryForZoom($dbUser, $zoomIntegration->getZoomCategory(), $emails, $meetingId);
+		kJobsManager::addImportJob(null, $entryId, $dbUser->getPartnerId(), $url);
+		KalturaLog::info('Zoom - upload entry to kaltura started, partner id: '. $zoomIntegration->getPartnerId() . 'host email: ' . $hostEmail . 'emails: ' . print_r($emails, true) .
+		'meeting Id: ' . $meetingId . 'entry Id: ' . $entryId);
+	}
+
+	/**
+	 * @param kuser $dbUser
+	 * @param string $zoomCategory
+	 * @param $emails
+	 * @param $meetingId
+	 * @return string
+	 * @throws Exception
+	 */
+	private function createEntryForZoom($dbUser, $zoomCategory, $emails, $meetingId)

```

Comment:
In my opinion, it would be better to move all these private functions to a helper class if possible.
Generally, the service class contains only the API calls that are getting exposed and the logic is moved to a class that exposes static functions which you can call.